### DISCUSSION TITLE
Content Cipher

### DISCUSSION THREAD

**Discussion by rfjakob** [encryption-alliance/unified-vault-format#12] [Approved]
Backstory for https://github.com/cryptomator/new-encryption-format/discussions/11 is that there may be multiple ciphers we want to support. Which ones?

**Comment by rfjakob** [comment]
I think we should only consider ciphers that also guarantee itegrity, i.e. authenticated encryption. 

IMO most interesting are:

1. AES-256-GCM because it's widely supported and often hardware-accelerated. Default cipher in gocryptfs.
2. ChaCha20-Poly1305 because it's fast even without hardware support. Only cipher in WireGuard.

Other ciphers that are less interesting:

1. XChaCha20-Poly1305. Less widely supported, not needed when we have a per-file encryption key. Supported in gocryptfs instead of ChaCha20-Poly1305 because gocryptfs does not have per-file encryption keys.

**Comment by overheadhunter** [comment]
> we should only consider ciphers that also guarantee itegrity, i.e. authenticated encryption

ack

I would add that ciphers should be reasonably common (well-tested, well-supported in wide spread libs).

Regarding AES-GCM, I would make 96 bit nonces and 128 bit tag size a requirement. We have discussed this a few years ago already, I will amend the rationale for this decision later.

**Reply by rfjakob** [reply]
Ad nonce and tag size: Ack, that's also what the Go stdlib uses per default ( https://pkg.go.dev/crypto/cipher#NewGCM ).

**Comment by overheadhunter** [comment]
How about starting with ONE cipher at first and adding further options with updated releases of this spec?

This allows a software to be compatible with this standard Version 1 and incrementally extend support for V2, V3, ...

**Reply by rfjakob** [reply]
Works for me.

**Reply by overheadhunter** [reply]
Drafted in #19
